### PR TITLE
API - Statistics

### PR DESCRIPTION
**Pull request by hungbkdn**

### CODE REVIEW

File: app/Http/Controllers/Api/StatisticController.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    /**
+     * Get count of resources.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function countResources()
+    {
+        $modelFiles = \File::files('../app/');
+
+        foreach ($modelFiles as $modelFile) {
```
Comment (ghost) [inline app/Http/Controllers/Api/StatisticController.php]: You don't need to get all Model file and loop all. Just get data for Model you want.
```php
$counts['foods'] = Food::count();
$counts['materials'] = Material::count();
```

File: app/Http/Controllers/Api/StatisticController.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+use Illuminate\Http\Response;
+use App\Http\Controllers\Controller;
+use App\Material;
+use App\Food;
+use App\OrderItem;
+
+class StatisticController extends ApiController
+{
+    /**
+     * Get count of resources.
```
Comment (ghost) [inline app/Http/Controllers/Api/StatisticController.php]: Please add more description. List all Model you count

File: app/Http/Controllers/Api/StatisticController.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        $columns = [
+            $tableName.'.id',
+            $tableName.'.name',
+            $tableName.'.category_id',
+            $tableName.'.price',
+            $tableName.'.image',
+            $tableName.'.description',
+            'count(*) as total_order'
+        ];
+        return OrderItem::selectRaw(implode(' ,', $columns))
```
Comment (ghost) [inline app/Http/Controllers/Api/StatisticController.php]: We can't use `select` and an array parameter?

File: app/Http/Controllers/Api/StatisticController.php [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    /**
+     * Get top of order by table type
+     *
+     * @param string $tableType type of target table
+     *
+     * @return array
+     */
+    public function getTopOrderOf($tableType)
```
Comment (ghost) [inline app/Http/Controllers/Api/StatisticController.php]: Please make this function into model `OrderItem `. And name should be `getTrendOrderByModel`.

File: app/Http/Controllers/Api/StatisticController.php [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            $tableName.'.image',
+            $tableName.'.description',
+            'count(*) as total_order'
+        ];
+        return OrderItem::selectRaw(implode(' ,', $columns))
+                        ->join($tableName, $tableName.'.id', '=', 'itemtable_id')
+                        ->groupBy(['itemtable_id', 'itemtable_type'])
+                        ->where('itemtable_type', $tableType)
+                        ->orderBy('total_order', 'desc')
+                        ->take(5)
```
Comment (ghost) [inline app/Http/Controllers/Api/StatisticController.php]: Please define `5` in Model as a `const`

File: app/Http/Controllers/Api/StatisticController.php [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+     * Get trends for foods and materials
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function getTrends()
+    {
+        $topFoods = $this->getTopOrderOf('App\\Food');
+        $topMaterials = $this->getTopOrderOf('App\\Material');
+
+        if ($topFoods && $topMaterials) {
```
Comment (ghost) [inline app/Http/Controllers/Api/StatisticController.php]: `if ($topFoods && $topMaterials)`
The `Trends` only display when has both of `Food` and `Meterial`?

**Review by ghost — COMMENTED** [review]
Please check my comment

**Comment by hungbkdn** [comment]
I have updated, please check it again!
Thank you so much

**Comment by ghost** [comment]
LGTM

**Pull request merged**